### DISCUSSION TITLE
Issue on up -d

### DISCUSSION THREAD

**Discussion by LfJohnVo** [serversideup/docker-php#184] [Q&A] [answered]
Have the following issue on the up -d proccess on laravel project:

Error: PHP Warning: require(/var/Www/html/vendor/autoload.php) failed to open stream

here is my docker-compose: 

```
version: "3.8"
services:
  # nginx:
  #   image: nginx:stable-alpine
  #   container_name: nginx
  #   restart: unless-stopped
  #   tty: true
  #   ports:
  #     - 80:80
  #   volumes:
  #     - .:/var/www/html
  #     - ./infra/nginx/default.conf:/etc/nginx/conf.d/default.conf
  #   depends_on:
  #     - php
  #     - postgres
  #   networks:
  #     - tabantajnetwork

  postgres:
    image: postgres:13-alpine
    container_name: postgres
    restart: unless-stopped
    tty: true
    volumes:
          - "./infra/postgres/data:/var/lib/postgresql/data"
    environment:
          - POSTGRES_USER=homestead
          - POSTGRES_PASSWORD=secret
          - POSTGRES_DB=homestead
    ports:
          - "5432:5432"
    networks:
          - tabantajnetwork

  php:
    # build:
    #   context: .
    #   dockerfile: Dockerfile
    image: serversideup/php:8.2-fpm-nginx
    container_name: php
    restart: unless-stopped
    tty: true
    volumes:
      - .:/var/www/html
    ports:
      - 80:80
    environment:
      SSL_MODE: "off"
    networks:
      - tabantajnetwork

networks:
  tabantajnetwork:
```

**Comment by jaydrogers** [comment]
Thanks for posting your code.

When are you exactly getting this error? Running the command from above gives me the expected results

<img width="2560" alt="image" src="https://github.com/serversideup/docker-php/assets/3174134/3d7d10f8-992c-41d5-94df-0a643c895381">

**Reply by LfJohnVo** [reply]
literaly when i use docker compose up. the docker-compose its on the root in a laravel project

**Reply by jaydrogers** [reply, marked as answer]
It's probably a permissions issue.

When you use `docker run` to run a command, it's going to run as `root` due to how PHP-FPM needs root to run.

You should switch to `webuser` instead.

More details here:
1. https://github.com/serversideup/docker-php/issues/179#issuecomment-1587430069
2. https://www.baeldung.com/linux/docker-image-container-switch-user
